stdpopsim/slim_engine.py

```diff
@@ -633,16 +704,32 @@ def script_file_f():
             recap_epoch = slim_makescript(
                     script_file, ts_file.name,
                     demographic_model, contig, samples,
-                    slim_scaling_factor, slim_burn_in, verbosity)
+                    slim_scaling_factor, slim_burn_in)
 
             script_file.flush()
 
             if not run_slim:
                 return None
 
             slim_cmd.append(script_file.name)
-            stdout = subprocess.DEVNULL if verbosity < 2 else None
-            subprocess.check_call(slim_cmd, stdout=stdout)
+            with subprocess.Popen(
+                    slim_cmd, bufsize=1, universal_newlines=True,
```

Comment:
I thought on reading this that you had a buffer of size 1 (which sounded like a bad idea), but looking it up you're doing exactly the right thing. Perhaps a quick comment saying "# bufsize=1 means line buffering " or something?

Unusual for Python to be so cryptic!